crates/matrix-sdk-crypto/src/olm/account.rs

```diff
@@ -829,16 +832,15 @@ impl ReadOnlyAccount {
         master_key: MasterPubkey,
     ) -> Result<SignatureUploadRequest, SignatureError> {
         let public_key =
-            master_key.get_first_key().ok_or(SignatureError::MissingSigningKey)?.to_string();
+            master_key.get_first_key().ok_or(SignatureError::MissingSigningKey)?.into();
         let mut cross_signing_key: CrossSigningKey = master_key.into();
         cross_signing_key.signatures.clear();
         self.sign_cross_signing_key(&mut cross_signing_key).await?;
 
-        let mut signed_keys = BTreeMap::new();
-        signed_keys
-            .entry(self.user_id().to_owned())
-            .or_insert_with(BTreeMap::new)
-            .insert(public_key, serde_json::to_value(cross_signing_key)?);
+        let mut user_signed_keys = SignedKeys::new();
+        user_signed_keys.add_cross_signing_keys(public_key, Raw::new(&cross_signing_key)?);
+
+        let signed_keys = BTreeMap::from_iter([(self.user_id().to_owned(), user_signed_keys)]);
```

Comment:
A bunch of places already use this construct but use `BTreeMap::from()` instead. Let's stick to `from()` for consistency sake.

Reply:
Oh, so we can actually just use `.into()` here because the type will be inferred from context.